src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -225,17 +231,42 @@ class Peer(
     init {
         logger.info { "initializing peer" }
         launch {
-            watcher.client.notifications.filterIsInstance<HeaderSubscriptionResponse>()
-                .collect { msg ->
-                    currentTipFlow.value = msg.blockHeight to msg.header
+            when (client) {
+                is IElectrumClient -> client.notifications.filterIsInstance<HeaderSubscriptionResponse>()
+                    .collect { msg ->
+                        currentTipFlow.value = msg.blockHeight to msg.header
+                    }
+                is MempoolSpaceClient -> while (true) {
+                    runCatching {
+                        client.getBlockTipHeight()?.let { currentBlockHeight ->
+                            logger.debug { "current block height is $currentBlockHeight" }
+                            currentTipFlow.value = currentBlockHeight to BlockHeader(0, BlockHash(ByteVector32.Zeroes), ByteVector32.Zeroes, 0, 0, 0)
```

Comment:
Note that we should get rid of the block header entirely, otherwise any code that relies on it won't work when using a `mempool.space` instance.

Reply:
Right, done in a325366b.